cranelift/assembler/meta/src/generate/inst.rs

```diff
@@ -31,29 +32,70 @@ impl dsl::Inst {
         format!("{}_{}", self.name.to_lowercase(), self.format.name.to_lowercase())
     }
 
-    /// `<inst>(<inst>),`
+    #[must_use]
+    fn immediate_name(&self) -> String {
+        self.format
+            .operands_by_kind()
+            .iter()
+            .find(|op| match op {
+                dsl::format::OperandKind::Imm(_) => true,
+                _ => false,
+            })
+            .map_or_else(
+                || String::new(),
+                |op| match op {
+                    dsl::format::OperandKind::Imm(dsl::Location::imm8) => "_ib".to_string(),
+                    dsl::format::OperandKind::Imm(dsl::Location::imm16) => "_iw".to_string(),
+                    dsl::format::OperandKind::Imm(dsl::Location::imm32) => "_id".to_string(),
+                    _ => "".to_string(),
+                },
+            )
+    }
+
+    /// `<inst>_<immediate_name>(<inst>),`
     pub fn generate_enum_variant(&self, f: &mut Formatter) {
         let variant_name = self.struct_name();
-        let struct_name = self.struct_name();
-        fmtln!(f, "{variant_name}({struct_name}),");
+        let immediate_name = self
+            .format
+            .operands_by_kind()
+            .iter()
+            .find(|op| match op {
+                dsl::format::OperandKind::Imm(_) => true,
+                _ => false,
+            })
+            .map_or_else(
+                || String::new(),
+                |op| match op {
+                    dsl::format::OperandKind::Imm(dsl::Location::imm8) => "_ib".to_string(),
+                    dsl::format::OperandKind::Imm(dsl::Location::imm16) => "_iw".to_string(),
+                    dsl::format::OperandKind::Imm(dsl::Location::imm32) => "_id".to_string(),
+                    _ => "".to_string(),
+                },
+            );
+
+        let struct_name = self.struct_name() + &immediate_name;
+        fmtln!(f, "{variant_name}{immediate_name}({struct_name}),");
```

Comment:
Let's not change the name of the enum variant... it should just match the name of the struct. Let's instead continue to enforce that the struct name is unique, which makes everything simpler. (Or is there something I don't understand here?)